libc/stat.c

```diff
@@ -0,0 +1,987 @@
+TEST_GROUP(stat_errno);
+
+
+TEST_SETUP(stat_mode)
+{
+	/* Removing/clearing elements that could affect test case results */
+	remove(path);
+	remove(symPath);
+	remove(tempPath);
+}
+
+
+TEST_TEAR_DOWN(stat_mode)
+{
+}
+
+
+TEST(stat_mode, id_mode_none)
+{
+	struct stat buffer;
+
+	fd = open(path, O_CREAT, 00000);
+
+	TEST_ASSERT_EQUAL_INT(0, stat(path, &buffer));
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, lstat(path, &buffer));
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, fstat(fd, &buffer));
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_FALSE(buffer.st_mode & S_ISGID);
+
+	close(fd);
+	remove(path);
+}
+
+
+TEST(stat_mode, id_mode_GID)
+{
+/* Disabled because of #743 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/743 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#743 issue");
+#endif
+	struct stat buffer;
+
+	fd = open(path, O_CREAT, S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, stat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, lstat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, fstat(fd, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	close(fd);
+	remove(path);
+}
+
+
+TEST(stat_mode, id_mode_UID)
+{
+/* Disabled because of #743 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/743 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#743 issue");
+#endif
+	struct stat buffer;
+
+	fd = open(path, O_CREAT, S_ISUID);
+
+	TEST_ASSERT_EQUAL_INT(0, stat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+
+	TEST_ASSERT_EQUAL_INT(0, lstat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+
+	TEST_ASSERT_EQUAL_INT(0, fstat(fd, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+
+	close(fd);
+	remove(path);
+}
+
+
+TEST(stat_mode, id_mode_UID_GID)
+{
+/* Disabled because of #743 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/743 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#743 issue");
+#endif
+	struct stat buffer;
+
+	fd = open(path, O_CREAT, S_ISUID | S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, stat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, lstat(path, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	TEST_ASSERT_EQUAL_INT(0, fstat(fd, &buffer));
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISUID);
+	TEST_ASSERT_TRUE(buffer.st_mode & S_ISGID);
+
+	close(fd);
+	remove(path);
+}
+
+
+TEST(stat_mode, vtx_mode)
+{
+/* Disabled because of #743 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/743 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#743 issue");
+#endif
+	struct stat buffer;
+
+	fd = open(path, O_CREAT, 00000);
```

Comment:
This is not possible because the file is deleted and created every test case. With the current idea, there will be one test case to check if the stat reads the file correctly for the zero permissions.